cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/LoopScope.kt

```diff
@@ -38,7 +38,27 @@ class LoopScope(loopStatement: Statement) :
     var starts: List<Node> = ArrayList()
```

Comment:
While you are at it: Could you remove the `starts()` function and move the logic into the getter only property `starts`, similar to `conditions`?

Reply:
Done in cd9ea0b140253e541c004c082bb9407671c9d8a2. I also changed the `when` expression slightly to enable smart casting and removed all explicit casts.